### PR TITLE
docs: add prebuilds

### PR DESCRIPTION
**Pull request by stefanicjuraj**
Fixes https://github.com/daytonaio/docs/issues/123

### CODE REVIEW

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
can we put PREBUILDS as their own menu item? as I would search for PREBUILDS but not PROJECTS. @Tpuljak ?

File: src/content/docs/usage/prebuilds.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Prebuilds
+description: Learn how to manage and reuse Prebuilds across Workspaces.
+sidebar:
+  label: Prebuilds
+---
+
+Prebuilds are designed to speed up development by setting up a [project configuration](/usage/projects#project-configuration) that listens to changes in the connected Git Provider repository. This ensures a quicker project creation process when initiating a Workspace.
```
Comment (ivan-burazin) [inline src/content/docs/usage/prebuilds.mdx:8]: > Prebuilds are designed to speed up development by setting up a [project configuration](/usage/projects#project-configuration) that listens to changes in the connected Git Provider repository. This ensures a quicker project creation process when initiating a Workspace.

Prebuilds are designed to accelerate development by pre-building your workspace. When you start a workspace, the prebuilt environment is used, ensuring that the workspace is initiated almost instantly.

Prebuilds achieve this by setting up a [project configuration](https://github.com/usage/projects#project-configuration) that monitors changes in the connected Git provider's repository.

File: src/content/docs/usage/prebuilds.mdx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+Prebuilds work by registering a listener to webhook events from the Git Provider by sending a public API endpoint that the Git Provider will use to send requests.
```
Comment (ivan-burazin) [inline src/content/docs/usage/prebuilds.mdx:10]: > Prebuilds work by registering a listener to webhook events from the Git Provider by sending a public API endpoint that the Git Provider will use to send requests.

Prebuilds work by registering a listener for webhook events from the Git provider. A public API endpoint is provided, which the Git provider uses to send these requests.

File: src/content/docs/usage/prebuilds.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## Add a Prebuild
+
+1. Run the following command to add a new Prebuild:
+
+   ```shell
+   daytona prebuilds add
+   ```
+
+   Using the command opens a view that prompts you to select a project configuration and the Git branch you plan to work on. You can then decide on a commit interval after which a build should be triggered, as well as any specific trigger files whose changes should immediately start the build process.
```
Comment (ivan-burazin) [inline src/content/docs/usage/prebuilds.mdx:20]: @stefanicjuraj @Tpuljak what is the unit of an interval? we should explain this

Comment (ivan-burazin) [inline src/content/docs/usage/prebuilds.mdx:20]: @stefanicjuraj @Tpuljak files? how do I define the files? just name? github path? what is the format, this is unclear to me

**Review by ivan-burazin — CHANGES_REQUESTED** [review]
Comments inline but believe some data is missing such as; Enabling / disabling, debuging, Rerun ...

File: src/content/docs/usage/prebuilds.mdx (hunk 3 above)
Comment (Tpuljak) [reply src/content/docs/usage/prebuilds.mdx:22]: 1 unit = 1 commit.

We can just put `(e.g. every 5 commits)`

Comment (Tpuljak) [reply src/content/docs/usage/prebuilds.mdx:22]: > Enter full paths for files whose changes you want to explicitly trigger a prebuild. Use new lines for multiple entries.

This is written below. I wouldn't explain more here.

File: src/content/docs/usage/prebuilds.mdx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+Subsequent `daytona create` calls will automatically detect the most recent existing build and use it to create the project.
+
+## List Prebuilds
+
+Daytona allows you to list all Prebuilds, providing you with an overview of the properties stored within each Prebuild configuration.
+
+1. Run the following command to list all Prebuilds:
+
+   ```shell
+   daytona build list
```
Comment (Tpuljak) [inline src/content/docs/usage/prebuilds.mdx:43]: `daytona prebuild list`

`daytona build`  commands are for managing builds.

Builds are artifacts of prebuild configuration. 
E.g. You define a commit interval of 5 in your prebuild configuration. Every 5 commits, a new build of the project will be produced and used as a cache when creating a workspace.

Also, users can run builds manually, without waiting for the prebuild condition to be met (`daytona build run`).

File: src/content/docs/usage/prebuilds.mdx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+Prebuilds are designed to speed up development by setting up a [project configuration](/usage/projects#project-configuration) that listens to changes in the connected Git Provider repository. This ensures a quicker project creation process when initiating a Workspace.
+
+Prebuilds work by registering a listener to webhook events from the Git Provider by sending a public API endpoint that the Git Provider will use to send requests.
+
+## Add a Prebuild
+
+1. Run the following command to add a new Prebuild:
+
+   ```shell
+   daytona prebuilds add
```
Comment (Tpuljak) [inline src/content/docs/usage/prebuilds.mdx:19]: Add a note for the `--run` flag that will auto-run the build when created.

**Review by Tpuljak — CHANGES_REQUESTED** [review]

**Comment by ivan-burazin** [comment]
Everything regarding running a pre-build is missing cc @idagelic to provide you with more info. Also seems prebuild is a prerequisite so that should be mentioned as well and that you need to setup a project for the repo of the prebuild

File: src/content/docs/usage/prebuilds.mdx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+   Retention
+   Maximum number of resulting builds stored at a time
+   >
+   ```
+
+Subsequent `daytona create` calls will automatically detect the most recent existing build and use it to create the project.
+
+## Use a Prebuild
+
+Once you added a Prebuild, it becomes active immediately based on the [project configuration](usage/projects#project-configuration) you provided. The Prebuild will automatically run whenever the specified conditions are met, such as when the defined number of commits is reached. For instance, if you specified a commit interval of `5`, the Prebuild will run after every `5` commits to the repository.
```
Comment (idagelic) [inline src/content/docs/usage/prebuilds.mdx:60]: Populate the "Trigger files" field with files whose changes you want to immediately trigger a prebuild without having to wait for the commit interval. Use the "Retention" field to define how many successful builds you want Daytona to remember (defaults to 3).

^I'd add this to the end

File: src/content/docs/usage/prebuilds.mdx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+As soon as you add a Prebuild, a build process will automatically start in the background unless you have configured it otherwise. The initial build process may take some time to complete, but subsequent builds will be significantly faster. However, if you try to create a Workspace before this build is completed, Daytona will start a fresh build, which might take longer.
```
Comment (idagelic) [inline src/content/docs/usage/prebuilds.mdx:62]: Replace:

`However, if you try to create a Workspace before this build is completed, Daytona will start a fresh build, which might take longer.`

with:

```
You can view the initial build's progress by running `daytona builds list`.
(table of daytona build list)
```

File: src/content/docs/usage/prebuilds.mdx [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+It is recommended to wait (e.g., 10 minutes) for the initial build to complete before creating a Workspace.
```
Comment (idagelic) [inline src/content/docs/usage/prebuilds.mdx:64]: Replace:
```
It is recommended to wait (e.g., 10 minutes) for the initial build to complete before creating a Workspace.

You can now create a Workspace using the Prebuild by running the following command:

```shell
daytona create
```

with:

Once it's finished, you can try running 
```shell
daytona create [PROJECT_CONFIG_NAME]
```

File: src/content/docs/usage/prebuilds.mdx [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+You can now create a Workspace using the Prebuild by running the following command:
+
+```shell
+daytona create
+```
+
+Upon running this command, Daytona will detect the most recent Prebuild associated with the selected project configuration. The Workspace will then use this Prebuild, significantly speeding up the setup process.
```
Comment (idagelic) [inline src/content/docs/usage/prebuilds.mdx:72]: I'd add "by utilizing the cached environment." to the end of this sentence and remove the one after it ("During... slowing down the workspace creation")

**Review by idagelic — COMMENTED** [review]
Suggested some changes for a more optimistic perspective when using a prebuild instead of cautions

File: src/content/docs/usage/prebuilds.mdx [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+## Add a Prebuild
+
+1. Run the following command to add a new Prebuild:
+
+   ```shell
+   daytona prebuilds add
+   ```
+
+   <Aside type="note">
+      Add a `--run` flag to the command to automatically run the build after creating the Prebuild.
```
Comment (idagelic) [inline src/content/docs/usage/prebuilds.mdx:25]: This Aside is now deprecated with the newest version, we can remove it

**Review by idagelic — APPROVED** [review]

**Review by Tpuljak — APPROVED** [review]

**Pull request merged**